azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -331,78 +338,211 @@ def __init__(self, xml_text):
         logger.verbose("Extension config shows status blob type as [{0}]", self.status_upload_blob_type)
 
     @staticmethod
-    def _parse_plugin(plugin):
-        ext_handler = ExtHandler()
-        ext_handler.name = getattrib(plugin, "name")
-        ext_handler.properties.version = getattrib(plugin, "version")
-        ext_handler.properties.state = getattrib(plugin, "state")
+    def _parse_plugin(ext_handler, plugin):
+
+        def _raise_config_error_if_none(attr_name, value):
+            if value is None:
+                raise ExtensionConfigError("{0} is None for ExtensionConfig, failing Extension.".format(attr_name))
```

Comment:
These values are expected to be there for every Plugin. If they're not there then we might (or not even) report wrong status while CRP would be polling for something else. Failing here to ensure we fail fast